src/netconf_confirmed_commit.c

```diff
@@ -0,0 +1,938 @@
+            ERR("Failed restoring backup for module: \"%s\".", module->name);
+            rename_failed_file(module_name, path);
+            continue;
+        }
+        if (unlink(path) == -1) {
+            ERR("Failed removing backup file \"%s\"", path);
+            goto cleanup;
+        }
+    }
+
+    if (unlink(NP2SRV_DIR RESTORE_FILE) == -1) {
+        ERR("Failed removing confirmed commit restore file.");
+    }
+
+cleanup:
+    free(user);
+    free(group);
+    free(path);
+    free(module_name);
+}
+
+/**
+ * @brief Remove all the backup files not marked as failed.
+ */
+static void
+clean_backup_directory(void)
+{
+    DIR *dir;
+    struct dirent *dirent;
+    char *suffix, *path;
+
+    dir = opendir(NP2SRV_DIR);
+    if (!dir) {
+        ERR("Could not open netopeer2 server directory: \"%s\"", NP2SRV_DIR);
+    }
+    while ((dirent = readdir(dir))) {
+        if (!strcmp(dirent->d_name, ".") || !strcmp(dirent->d_name, "..") || !strcmp(dirent->d_name, "restore")) {
+            continue;
+        }
+
+        suffix = strstr(".failed", dirent->d_name);
+        if (suffix) {
+            continue;
+        }
+
+        if (asprintf(&path, "%s/%s", NP2SRV_DIR, dirent->d_name) == -1) {
+            EMEM;
+            return;
+        }
+
+        if (unlink(path) == -1) {
+            ERR("Could not remove backup file \"%s\"", path);
+            free(path);
+            return;
+        }
+        free(path);
+    }
+}
+
+static void
+ncc_disarm_timer(void)
+{
+    pthread_mutex_lock(&commit_ctx.lock);
+    timer_delete(commit_ctx.timer);
+    commit_ctx.timer = NULL;
+    clean_backup_directory();
+    pthread_mutex_unlock(&commit_ctx.lock);
+}
+
+static void
+ncc_trigger_timer(void)
+{
+    changes_rollback((union sigval)0);
+    ncc_disarm_timer();
+}
+
+/**
+ * @brief check if NP2SRV_DIR exists, try creating it otherwise
+ *
+ * @return SR_ERR_SYS When failed during creation of the directory.
+ * @return SR_ERR_OK When directory exists or was successfully created.
+ */
+static int
+check_dir(void)
+{
+    int rc = SR_ERR_OK;
+    const char *path = NP2SRV_DIR;
+    struct stat buf;
+
+    printf("%s", path);
+    if (stat(NP2SRV_DIR, &buf) == -1) {
+        VRB("Creating directory \"%s\".", NP2SRV_DIR);
+        if (mkdir(NP2SRV_DIR, S_IRWXU) == -1) {
+            ERR("Failed creating directory \"%s\".", NP2SRV_DIR);
+            rc = SR_ERR_SYS;
+        }
+    }
+    return rc;
+}
+
+/**
+ * @brief Backup a module into a file
+ *
+ * @param[in] session Sysrepo session used to get data of the module.
+ * @param[in] module Module to backup.
+ *
+ * @return SR_ERR_LY When printing into the file failed.
+ * @return SR_ERR_NO_MEMORY When memory ran during allocation
+ * @return SR_ERR_OK On success
+ */
+static int
+backup_module(sr_session_ctx_t *session, struct lys_module *module)
+{
+    int rc = SR_ERR_OK;
+    char *path = NULL, *xpath = NULL;
+    struct lyd_node *node = NULL;
+
+    /* LOCK commit_ctk.lock */
```

Comment:
The only instance of you trying to use the comments so decide whether you want to use them or not, seems not necessary in this file. In any case, something like `COMMIT CTX (UN)LOCK` should probably be the message.